### DISCUSSION TITLE
Hue 4.10 build shows distutils-related errors with Python 3.8

### DISCUSSION THREAD

**Discussion by SirYaro** [cloudera/hue#2720] [Questions]
Hi
I'm having problem building hue 4.10 on my test box.
```
[yaro@yaro-42 ~]$ cat /etc/redhat-release 
AlmaLinux release 8.5 (Arctic Sphynx)
[yaro@yaro-42 ~]$ 
```
I can easily build dev version with:
```
git clone https://github.com/cloudera/hue.git # or by taking the source from https://cdn.gethue.com/downloads/hue-4.10.0.tgz
cd hue
export PYTHON_VER=python3.8
make apps
```
, and get hue working within 15 minutes.

All (but one) deps listed here `https://docs.gethue.com/administrator/installation/dependencies/#centosrhel` are installed 
```
sudo yum install ant asciidoc cyrus-sasl-devel cyrus-sasl-gssapi cyrus-sasl-plain gcc gcc-c++ krb5-devel libffi-devel libxml2-devel libxslt-devel make mysql mysql-devel openldap-devel python-devel sqlite-devel gmp-devel openssl-devel 
Ostatnio sprawdzono ważność metadanych: 0:34:53 temu w dniu śro, 16 lut 2022, 11:50:46.
[...]
Brak wyników dla parametru: python-devel
[...]
Błąd: Brak wyników: python-devel               # <-- Means, "Error: No results for python-devel"
[yaro@yaro-43 hue-4.10.0]$ 
```
But I have `python*-devel*` installed:
```
[yaro@yaro-43 hue-4.10.0]$ rpm -qa|grep python|grep devel
python38-devel-3.8.8-4.module_el8.5.0+2570+a865257a.x86_64
[yaro@yaro-43 hue-4.10.0]$ 
```
So the actual error I'm getting is not too obvious (to me):
```
[root@yaro-43 hue-4.10.0]# rm -rf /tmp/yaro/hue;PREFIX=/tmp/yaro/ make install
[...]
--- Building egg for docutils-0.14
Error: The "distutils" standard module, which is required for the 
installation of Docutils, could not be found.  You may need to 
install a package called "python-devel" (or similar) on your 
system using your package manager.
make[2]: *** [/root/hue-4.10.0/Makefile.sdk:133: /root/hue-4.10.0/desktop/core/build/docutils-0.14/egg.stamp] Błąd 1
make[2]: Opuszczenie katalogu '/root/hue-4.10.0/desktop/core'
make[1]: *** [Makefile:112: .recursive-install-bdist/core] Błąd 2
make[1]: Opuszczenie katalogu '/root/hue-4.10.0/desktop'
make: *** [Makefile:132: install-desktop] Błąd 2
[root@yaro-43 hue-4.10.0]#
```

I know very little about python and while error msg is not very descriptive, my guess is that either the installed *-dev package doesn't provide some required files, or it save them in some weird location. 

Could someone wiser help me troubleshot that? What am I missing here?

**Comment by Harshg999** [comment]
Hi @SirYaro, can you try installing `python3-distutils`?

**Reply by SirYaro** [reply]
@Harshg999 
Yes. I did. Almalinux seems to miss that package. I took python3-distutils-extra-2.39-16.el8.noarch from centos8, but it changed nothing at all.

**Comment by irudayambics** [comment]
Hello Everyone, I am getting the same error

/home/hue/hue-4.10.0# PREFIX=/home/hue/app make install
"PYTHON_VER is python3.8."
"Python 3 module install via pip"
"SYS_PYTHON is /usr/bin/python3.8."
"ENV_PYTHON is /home/hue/hue-4.10.0/build/env/bin/python3.8."
--- Installing core source structure...
--- Installing Desktop core...
INSTALL_DIR=/home/hue/app/hue make -C desktop install
make[1]: Entering directory '/home/hue/hue-4.10.0/desktop'
"PYTHON_VER is python3.8."
"Python 3 module install via pip"
"SYS_PYTHON is /usr/bin/python3.8."
"ENV_PYTHON is /home/hue/hue-4.10.0/build/env/bin/python3.8."
"----------------> ENV_INSTALL_TARGETS: .recursive-env-install/core .recursive-env-install/libs/aws .recursive-env-install/libs/azure .recursive-env-install/libs/hadoop .recursive-env-install/libs/indexer .recursive-env-install/libs/libanalyze .recursive-env-install/libs/liboauth .recursive-env-install/libs/liboozie .recursive-env-install/libs/librdbms .recursive-env-install/libs/libsaml .recursive-env-install/libs/libsentry .recursive-env-install/libs/libsolr .recursive-env-install/libs/libzookeeper .recursive-env-install/libs/metadata .recursive-env-install/libs/notebook .recursive-env-install/libs/dashboard .recursive-env-install/libs/kafka"
INSTALL_DIR=/home/hue/app/hue/desktop/core \
        INSTALL_CONF_DIR=/home/hue/app/hue/desktop/conf \
        make -C core install-bdist
make[2]: Entering directory '/home/hue/hue-4.10.0/desktop/core'
"PYTHON_VER is python3.8."
"Python 3 module install via pip"
"SYS_PYTHON is /usr/bin/python3.8."
"ENV_PYTHON is /home/hue/hue-4.10.0/build/env/bin/python3.8."
--- Building egg for docutils-0.14
Error: The "distutils" standard module, which is required for the
installation of Docutils, could not be found.  You may need to
install a package called "python-devel" (or similar) on your
system using your package manager.
make[2]: *** [/home/hue/hue-4.10.0/Makefile.sdk:133: /home/hue/hue-4.10.0/desktop/core/build/docutils-0.14/egg.stamp] Error 1
make[2]: Leaving directory '/home/hue/hue-4.10.0/desktop/core'
make[1]: *** [Makefile:112: .recursive-install-bdist/core] Error 2
make[1]: Leaving directory '/home/hue/hue-4.10.0/desktop'
make: *** [Makefile:132: install-desktop] Error 2
